#! /bin/sh /usr/share/dpatch/dpatch-run
## 60-getweb.dpatch by Luca Bedogni <me@lucabedogni.it>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Improve getweb also for installing the firmware

@DPATCH@
--- foo2zjs_20070718/getweb.in.orig	2007-06-05 17:01:14.000000000 +0200
+++ foo2zjs_20070718/getweb.in	2007-09-22 21:51:59.000000000 +0200
@@ -63,6 +63,18 @@
 fi
 
 #
+#   Put firmware in right place
+putfw() {
+    inputname=$1
+    outputname=$2
+    if [ $UID -ne 0 ]; then
+        echo "Please, type root's password..."
+        su -c "arm2hpdl $inputname > /usr/share/foo2zjs/firmware/$outputname"
+        exit
+    fi
+}
+
+#
 #	Download a .EXE file from the web, unzip it, and extract the
 #	files we want
 getexe() {
@@ -96,6 +108,8 @@
 	gettgz \
 	    http://foo2zjs.rkkda.com sihp1000.tar.gz \
 	    ""
+	putfw sihp1000.img sihp1000.dl
+	rm -f sihp1000.img
 	#echo "provided"
 	#getexe \
 	    #ftp://ftp.hp.com/pub/softlib/software1/lj1488/lj-1145-2 \
@@ -106,6 +120,8 @@
 	gettgz \
 	    http://foo2zjs.rkkda.com sihp1005.tar.gz \
 	    ""
+	putfw sihp1005.img sihp1005.dl
+	rm -f sihp1005.img
 	#echo "provided"
 	#getexe \
 	    #ftp://ftp.hp.com/pub/softlib/software2/COL2222/lj-10067-2 \
@@ -116,11 +132,15 @@
 	gettgz \
 	    http://foo2zjs.rkkda.com sihp1018.tar.gz \
 	    ""
+	putfw sihp1018.img sihp1018.dl
+	rm -f sihp1018.img
 	;;
     lj1020|1020)
 	gettgz \
 	    http://foo2zjs.rkkda.com sihp1020.tar.gz \
 	    ""
+	putfw sihp1020.img sihp1020.dl
+	rm -f sihp1020.img
 	;;
     2200dl|2200)
 	getexe \
